components/Buttons/tests/unit/ButtonsTests.m

```diff
@@ -1116,43 +1116,37 @@ - (void)testAdjustsFontProperty {
                              @"Font size should be equal to MDCFontTextStyleButton's.");
 }
 
-/**
- Test legacy dynamic type has no impact on a @c MDCButton when @c
- adjustFontForContentSizeCategoryWhenScaledFontIsUnavailable is set to @c NO before setting @c
- mdc_adjustsFontForContentSizeCategory to @c YES that the font stays the same.
- */
-- (void)testLegacyDynamicTypeDisabledThenDynamicTypeTurnedOn {
+- (void)testLegacyDynamicTypeDisabledDoesNotChangeTheFont {
   // Given
   UIFont *fakeFont = [UIFont systemFontOfSize:55];
   [self.button setTitleFont:fakeFont forState:UIControlStateNormal];
-  UIFont *originalFont = self.button.titleLabel.font;
   self.button.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = NO;
 
   // When
   self.button.mdc_adjustsFontForContentSizeCategory = YES;
 
   // Then
-  XCTAssertTrue([self.button.titleLabel.font mdc_isSimplyEqual:originalFont],
-                @"%@ is not equal to %@", self.button.titleLabel.font, originalFont);
+  XCTAssertTrue([[self.button titleFontForState:UIControlStateNormal] mdc_isSimplyEqual:fakeFont],
+                @"%@ is not equal to %@", [self.button titleFontForState:UIControlStateNormal],
+                fakeFont);
+  XCTAssertTrue([self.button.titleLabel.font mdc_isSimplyEqual:fakeFont], @"%@ is not equal to %@",
+                self.button.titleLabel.font, fakeFont);
 }
 
-/**
- Test legacy dynamic type impacts a @c MDCButton when @c
- adjustFontForContentSizeCategoryWhenScaledFontIsUnavailable is set to @c YES that the font changes.
- */
-- (void)testLegacyDynamicTypeEnabled {
+- (void)testLegacyDynamicTypeEnabledDoesChangeTheFont {
```

Comment:
Please explicitly set `self.button.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = YES;` in the `Given` section.  That will protect this test if the default value changes.